Conjur/CONTRIBUTING.md

```diff
@@ -118,7 +151,233 @@ important to follow when developing.
+  of the repo:
+  ```sh-session
+  $ git push origin v1.2.3
+  ```
+
+  > Note: Overwriting or removing old tags is extremely discouraged
+
+## GitHub Releases
+
+GitHub releases are the main way that our users consume some of our end products so it is very
+important that these are maintained 1:1 along with tags.
+
+> **Note: Only full releases are considered for inclusion in the Conjur OSS Suite**
+
+### Draft Release Creation
+
+Once the [tag](#tagging) has been pushed to upstream repository and you are ready to create an
+official release of that tagged version we can go to the GitHub UI to first create a draft release:
+- **Navigate to the releases page**. In GitHub, under `<main_repo_url>/releases` you can find
+  the list of all relases that are available for the repository. We will click the `Draft a new release`
+  in the top right to get us into the "new release" page.
+- **Fill out the relevant release info**. Make sure that:
+  - `Tag version` is the tag you created for this release.
+  - `Release title` matches exactly the tag name (`v<semver>`).
+  - `Describe this release` is populated with changelog entries for this tagged version.
+  - Any files that you may want included this release are attached. This typically includes the
+    `LICENSE.md` file, `CHANGELOG.md` file, binary executable (if any), SHA256SUM of the
+    executable artifacts, etc).
+- **Create the draft release**. Check the box `This is a pre-release` and click `Save Draft`
+  which will create a draft release that is not yet visible to the public.
+
+Once the draft release is created it's a good time to triple-check the release and ensure that
+all of the executable files pass a smoke test for their environments. While this is not strictly
+mandatory, it prevents issues with mis-labeled or incompatible binaries.
+
+### Pre-release Publishing
+
+With draft release smoke test completed, we can make a "pre-release" (non-production ready
+release) to let the users know that we have a new release but that we are performing a slow
+rollout in an effort to ensure low impact if any bugs are found in this new release. This
+stage may last as little or as much as needed and it's at the discretion of the maintainers
+but should be guided by thinks like: the size of user base, cascading impact for bugs, level
+of automated testing in the repo, risk of changes causing unintended problems, and other
+concerns.
+
+To make a pre-release:
+- **Go to the draft release page**. The draft release can be found under
+  `<main_repo_url>/releases` for your repo.
+- **Publish the pre-release**. Double-check that the `This is a pre-release` is selected and
+  click `Publish release`.
+
+### Release Publishing
+
+After some (project-dependent) period of time of usage by the users of the pre-released
+version of the software, the confidence that the software works as expected is much higher
+and we can then turn our pre-release into a full release:
```

Comment:
some projects may have specific requirements to go from prerelease to release

for example, we tag prereleases in secretless for new tags bc at the time of tagging, we've run through a comprehensive test suite and have confidence to make a release but haven't done stability / performance tests to say it's production ready yet

once we've reviewed the changes and either determined they're not going to impact performance or we've validated the performance via testing so that we have some assurance the release meets our [stability requirements](https://github.com/cyberark/secretless-broker/blob/master/README.md#stable-release-definition), _then_ we change the pre-release to a release


which is a very long winded way of saying that it's not necessarily "after a period of time", but it's a project-specific definition of production readiness requirements that must be met to turn a pre-release into a release